src/test/java/com/google/cloud/tools/appengine/operations/AppYamlProjectStagingTest.java

```diff
@@ -118,6 +118,60 @@ public void testStageArchive_standardPath() throws IOException, AppEngineExcepti
     verify(mock).stageStandardArchive(config);
   }
 
+  @Test
+  public void testStageArchive_java11StandardBinaryPath() throws IOException, AppEngineException {
+    config =
+        AppYamlProjectStageConfiguration.builder()
+            .appEngineDirectory(appEngineDirectory)
+            .artifact(temporaryFolder.newFile("myscript.sh").toPath())
+            .stagingDirectory(stagingDirectory)
+            .extraFilesDirectories(extraFilesDirectories)
+            .build();
+
+    Files.write(
+        appEngineDirectory.resolve("app.yaml"),
+        "runtime: java11\nentrypoint: anything\n".getBytes(StandardCharsets.UTF_8),
+        StandardOpenOption.CREATE_NEW);
+
+    // mock to watch internal calls
+    AppYamlProjectStaging mock = Mockito.mock(AppYamlProjectStaging.class);
+    Mockito.doCallRealMethod().when(mock).stageArchive(config);
+
+    mock.stageArchive(config);
+    verify(mock).stageStandardBinary(config);
+  }
+
+  @Test
+  public void testStageArchive_java11BinaryWithoutEntrypoint()
+      throws IOException, AppEngineException {
+    Path badArtifact = temporaryFolder.newFile("myscript.sh").toPath();
+    config =
+        AppYamlProjectStageConfiguration.builder()
+            .appEngineDirectory(appEngineDirectory)
+            .artifact(badArtifact)
+            .stagingDirectory(stagingDirectory)
+            .extraFilesDirectories(extraFilesDirectories)
+            .build();
+
+    Files.write(
+        appEngineDirectory.resolve("app.yaml"),
+        "runtime: java11\n".getBytes(StandardCharsets.UTF_8),
+        StandardOpenOption.CREATE_NEW);
+
+    AppYamlProjectStaging testStaging = new AppYamlProjectStaging();
+
+    try {
+      testStaging.stageArchive(config);
+      fail();
```

Comment:
use static import, since there was a lot of that in this file, although I dunno, I think our style has been pretty inconsistent in appengine-plugins-core